src/main/java/core/basesyntax/service/impl/TransactionParserServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.service.TransactionParserService;
+
+public class TransactionParserServiceImpl implements TransactionParserService {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int OPERATION_VALUE_INDEX = 2;
+    private final FruitTransactionService fruitTransactionService;
+
+    public TransactionParserServiceImpl(FruitTransactionService fruitTransactionService) {
+        this.fruitTransactionService = fruitTransactionService;
+    }
+
+    @Override
+    public void parseTransactions(String data) {
```

Comment:
```suggestion

    @Override
    public List<FruitTransaction> parseTransactions(String data) {
```

The responsibility of parser is to parse. it should not interact with other services